src/main/java/com/ssu/schedule/parser/ParserSSU.java

```diff
@@ -45,18 +45,19 @@
 
     private final FacultyRepository facultyRepository;
     private final GroupRepository groupRepository;
+    private final OkHttpClient client;
 
     private TimeList timeList = new TimeList();
 
     @Autowired
     public ParserSSU(FacultyRepository facultyRepository, GroupRepository groupRepository) {
         this.facultyRepository = facultyRepository;
         this.groupRepository = groupRepository;
+        this.client = new OkHttpClient();
```

Comment:
It seems you don't understand my comment )
Create method in spring java config, annotate it as @Bean, return configured http client
Autowire it here like repositories (add constructor param and etc.)